### DISCUSSION TITLE
Dask dask arrays

### DISCUSSION THREAD

**Discussion by sehHeiden** [corteva/rioxarray#669] [Q&A]
After some years of using rioxarray, I used the first times with the chunks options.
Because the raster had a matrix size of about 70k * 60 k elements.  For the first times something, that did not fit into the ram when computing on the raster.
I did have high hopes, because using multiple cores for one matrix and using less ram would be nice.

I tried loading with chunks set to:
-  'auto'
- (4, 4)
- {'x': 5, 'y': 5}, or higher, like 10, or 25 in each direction
With Client set to four or five workers.

In general the dask functions I used never finished:
1. gradient
2. multiplication --> arctan, or arctan2
3. rad2deg

Therefor, I tried to add a re-projection from the resolution of 1 meter to 5 meter.
This never finished, too.

I encountered the following problems:
- opening with chunks=(4, 4), opened with chunks=(4, y_length). I did not encounter that with the dict.
- often got a gc warning , 40+ %,  or 70+ %
- sometimes got a warning, that the size of the graph is to large.
- Instead of loading less data the huge matrix, was often loaded multiple times

Hence, I opened, without chunks, resized and than computed. Which worked in seconds instead of hours.

**Comment by snowman2** [comment]
Related: https://github.com/corteva/rioxarray/issues/119

**Reply by sehHeiden** [reply]
Although I ran in timing problems with and without re-projection.
